### PR TITLE
try out replacing Dsymbol.error() part 6

### PR DESCRIPTION
**Pull request by WalterBright**
more of https://github.com/dlang/dmd/pull/15657, in dsymbolsem.d

### CODE REVIEW

**Review by dkorpel — APPROVED** [review]

File: compiler/src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -2217,7 +2218,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
         if (ed.members.length == 0)
         {
-            ed.error("enum `%s` must have at least one member", ed.toChars());
+            .error(ed.loc, "%s `%s enum `%s` must have at least one member", ed.kind, ed.toPrettyChars, ed.toChars());
```
Comment (ibuclaw) [inline compiler/src/dmd/dsymbolsem.d:2221]: This introduced a diagnostic regression - there's an odd number of backticks present in the format string.

Comment (thewilsonator) [reply compiler/src/dmd/dsymbolsem.d:2221]: https://github.com/dlang/dmd/pull/17002

Comment (ibuclaw) [reply compiler/src/dmd/dsymbolsem.d:2221]: Codecov says this isn't covered by any file in the testsuite either.

**Comment by ibuclaw** [comment]
This PR introduced a regression https://issues.dlang.org/show_bug.cgi?id=24812

**Comment by thewilsonator** [comment]
https://github.com/dlang/dmd/pull/17004

**Pull request merged**